src/media/ros/ros_reader.cpp

```diff
@@ -1434,35 +1439,34 @@ namespace librealsense
         return std::make_pair(id, std::make_shared<const_value_option>(description, value));
     }
 
-    std::shared_ptr<librealsense::processing_block_interface> ros_reader::create_processing_block(const rosbag::MessageInstance& value_message_instance, bool& depth_to_disparity, std::shared_ptr<options_interface> options)
+    std::shared_ptr< librealsense::processing_block_interface >
+    ros_reader::create_processing_block( const rosbag::MessageInstance & value_message_instance,
+                                         bool & depth_to_disparity,
+                                         std::shared_ptr< options_interface > sensor_options )
     {
         auto processing_block_msg = instantiate_msg<std_msgs::String>(value_message_instance);
-        rs2_extension id;
-        convert(processing_block_msg->data, id);
-        std::shared_ptr<librealsense::processing_block_interface> disparity;
-
-        switch (id)
-        {
-        case RS2_EXTENSION_DECIMATION_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_DECIMATION_FILTER>::type>();
-        case RS2_EXTENSION_THRESHOLD_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_THRESHOLD_FILTER>::type>();
-        case RS2_EXTENSION_DISPARITY_FILTER:
-            disparity = std::make_shared<ExtensionToType<RS2_EXTENSION_DISPARITY_FILTER>::type>(depth_to_disparity);
-            depth_to_disparity = false;
-            return disparity;
-        case RS2_EXTENSION_SPATIAL_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_SPATIAL_FILTER>::type>();
-        case RS2_EXTENSION_TEMPORAL_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_TEMPORAL_FILTER>::type>();
-        case RS2_EXTENSION_HOLE_FILLING_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_HOLE_FILLING_FILTER>::type>();
-        case RS2_EXTENSION_HDR_MERGE:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_HDR_MERGE>::type>();
-        case RS2_EXTENSION_SEQUENCE_ID_FILTER:
```

Comment:
All other extensions are needed here anymore?
Is that backward compatible?

Reply:
Yes: the factory checks both extension names and block names. See the next comment...